src/vs/workbench/contrib/webview/browser/webview.ts

```diff
@@ -189,6 +189,11 @@ export interface IWebview extends IDisposable {
 	 */
 	readonly origin: string;
 
+	/**
+	 * The code window the webview is contained within.
+	 */
+	readonly codeWindow: CodeWindow;
```

Comment:
Would like to not expose this publicly if possible

Reply:
I now made it private by reading dom nodes when possible, this might be false positive sometimes though, let's test.